host/sampi/http.go

```diff
@@ -119,8 +140,10 @@ func (c *HTTPAPI) RegisterHost(w http.ResponseWriter, r *http.Request, ps httpro
 
 	c.Cluster.state.Begin()
 	if c.Cluster.state.HostExists(h.ID) {
+		err := errors.New("sampi: host exists")
+		l.Error("rolling back because of error", "err", err)
```

Comment:
This seems redundant because `HostExists` should log that the host exists, and `Rollback` should log that a rollback is happening.